src/Initializer/Parameters/ParameterReader.cpp

```diff
@@ -1,11 +1,56 @@
 #include "ParameterReader.h"
 #include <memory>
+#include <optional>
 #include <string>
 #include <utils/logger.h>
 #include <vector>
 
+#include "Common/filesystem.h"
+
 namespace seissol::initializer::parameters {
 
+ParameterReader::ParameterReader(const YAML::Node& node, const std::string& rootPath, bool empty)
+    : node(node), rootPath(rootPath), empty(empty) {}
+
+std::optional<std::string> ParameterReader::readPath(const std::string& field) {
+  const auto fileName = read<std::string>(field);
+  if (fileName.has_value()) {
+    const auto lastPath = filesystem::path(rootPath);
+    const auto nextPath = filesystem::path(fileName.value());
+    const auto loadFileName = [&]() {
+      if (nextPath.is_relative()) {
+        // remove file
+        return lastPath.parent_path() / nextPath;
+      } else {
+        return nextPath;
+      }
+    }();
+    // try to get the full file name (if that works)
+    if (filesystem::exists(loadFileName)) {
+      return filesystem::canonical(loadFileName);
+    } else if (filesystem::exists(nextPath)) {
+      return filesystem::canonical(nextPath);
+    } else {
+      // otherwise, just return the string (TODO: or should be fail here then?)
```

Comment:
We can do that here.

The only thing that currently still complicates that would be that the Netcdf files so far did not need a file ending.
However, iirc they are only ever used in case of convergence tests at this point in time (because they support periodic boundary conditions, and PUML doesn't—yet), so we could just enforce the file endings now. Or alternatively, supply an optional file ending to this function.